cmd/podman/common/completion.go

```diff
@@ -282,6 +284,61 @@ func getNetworks(cmd *cobra.Command, toComplete string, cType completeType) ([]s
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
 
+func getPathCompletion(root string, toComplete string) []string {
+	if toComplete == "" {
+		toComplete = "/"
+	}
+	// Important: securejoin is required to make sure we never leave the root mount point
+	userpath, err := securejoin.SecureJoin(root, toComplete)
+	if err != nil {
+		cobra.CompErrorln(err.Error())
+		return nil
+	}
+	var base string
+	f, err := os.Open(userpath)
+	if err != nil {
+		// Do not use path.Dir() since this cleans the paths which
+		// then no longer matches the user input.
+		userpath, base = path.Split(userpath)
+		toComplete, _ = path.Split(toComplete)
+		f, err = os.Open(userpath)
+		if err != nil {
+			return nil
+		}
+	}
+	stat, err := f.Stat()
+	if err != nil {
+		cobra.CompErrorln(err.Error())
+		return nil
+	}
+	if !stat.IsDir() {
+		// nothing to complete since it is no dir
+		return nil
```

Comment:
I didn't test it but wonder if we shouldn't return `[]string{userpath}` here so the file will be returned.

Reply:
There is no point in returning the same as the input. Also more important userpath is the full path on the host so it will be wrong.